Log browser.OpenURL errors, try some hardcoded fallback browsers on Windows

AlbinoDrought opened the pull request:
<!--
READ CAREFULLY: Before submitting your PR, please ensure you have created an issue for your PR.
It is essential that you do this so that we can discuss the proposed changes before you spend time on them.
If you do not create an issue, your PR may be rejected without review.
If a relevant issue already exists, please reference it in your PR by including `Fixes #<issue number>` in your PR description.
-->

# Description

In all platforms, logs errors returned by `browser.OpenURL`.

On Windows, tries the additional hardcoded browser paths:

- `\Program Files (x86)\Microsoft\Edge\Application\msedge.exe`
- `\Program Files\Google\Chrome\Application\chrome.exe`
- `\Program Files (x86)\Google\Chrome\Application\chrome.exe`
- `\Program Files\Mozilla Firefox\firefox.exe`

Fixes #3261 

See also https://discord.com/channels/1042734330029547630/1207762177277034546

## Type of change
  
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Windows
- [ ] macOS
- [ ] Linux

> I was able to update the software with the new version you created and got in without issue. Thank you very much! 
> - Customer with invalid default browser config
  
## Test Configuration

- Use Windows
- Set your default browser to Firefox (Settings -> Apps -> Default Apps -> Web Browser)
- Rename your firefox.exe to something.else

# Checklist:

- [ ] I have updated `website/src/pages/changelog.mdx` with details of this PR
- [ ] My code follows the general coding style of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Bug Fixes**
	- Improved error handling and implemented fallback mechanisms for opening URLs across different desktop platforms.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

leaanthony commented on v2/internal/frontend/desktop/darwin/browser.go:
```diff
@@ -10,5 +10,7 @@ import (
 	// Specific method implementation
-	_ = browser.OpenURL(url)
+	if err := browser.OpenURL(url); err != nil {
+		f.logger.Error(err.Error())
```
```suggestion
		f.logger.Error("Unable to open default system browser")
```

leaanthony commented on v2/internal/frontend/desktop/linux/browser.go:
```diff
@@ -8,5 +8,7 @@ import "github.com/pkg/browser"
 	// Specific method implementation
-	_ = browser.OpenURL(url)
+	if err := browser.OpenURL(url); err != nil {
+		f.logger.Error(err.Error())
```
```suggestion
		f.logger.Error("Unable to open default system browser")
```

leaanthony commented on v2/internal/frontend/desktop/windows/browser.go:
```diff
@@ -5,10 +5,32 @@ package windows
+	}
+	for _, fallback := range fallbackBrowserPaths {
+		if err2 := openBrowser(fallback, url); err2 == nil {
+			f.logger.Warning("default browser.OpenURL failed but fell back to %v, original err: %v", fallback, err.Error())
```
I'm not sure we need these. I'm wondering what sort of action would be taken based on seeing them.

AlbinoDrought replied:
I agree, removed in https://github.com/wailsapp/wails/compare/69375ca82904e0c32489f98ff6db6eb05ae542e8..cdeac67d3dc53df24178b92666fbb6b4b090d3ce`

leaanthony commented on v2/internal/frontend/desktop/windows/browser.go:
```diff
@@ -5,10 +5,32 @@ package windows
+			return
+		}
+	}
+	f.logger.Error("default browser.OpenURL failed, no fallback browser success, original err: %v", err.Error())
```
```suggestion
	f.logger.Error("Unable to open default system browser")
```

leaanthony reviewed (changes requested):
Thanks for doing this! I checked the browser module we use and I think the error returned by it is fairly useless so suggest we provide our own error messages.

AlbinoDrought commented:
(Sorry, not used to Github reviews)

Replaced error messages and removed warn on fallback: https://github.com/wailsapp/wails/compare/69375ca82904e0c32489f98ff6db6eb05ae542e8..cdeac67d3dc53df24178b92666fbb6b4b090d3ce

leaanthony commented:
Thanks 🙏 Please could you add an entry to the changelog located at `website/src/pages/changelog.mdx`?

leaanthony commented:
@AlbinoDrought - ping 😄
